src/content/blog/ddev-bundled-tools-using-custom-versions.md

```diff
@@ -0,0 +1,52 @@
+
+---
+title: "How-to downgrade terminus in DDEV and customize bundled tools"
+pubDate: 2025-06-12
+summary: How to upgrade/downgrade a utility provided by DDEV, or add a custom one for a given project
+author: Bill Seremetis
+featureImage:
+  src: /img/blog/2025/06/ddev-tool-install.png
+  alt: "Installing custom software packages in the containerized environment"
+  shadow: true
+categories:
+  - Guides
+  - DevOps
+---
+
+_This guest post is by DDEV community member and [Drupal](https://drupal.org)
+contributor [Bill Seremetis](/blog/author/bill-seremetis/) and sponsored by
+[Annertech](https://www.annertech.com)._
+
+DDEV comes bundled with a predefined set of tools, `terminus` being one of them.
+Latest releases of `terminus` are not compatible with older PHP versions, 
+thus there is a need to downgrade the package inside DDEV.
+
+This guide covers how and will also explain how to use the same technique to install
+custom tools too.
+
+## Manually downgrading Terminus
+
+We can specify which version to use on a given project by overriding the one
+provided by DDEV easily by using a custom Dockerfile:
+
+```dockerfile
+# .ddev/web-build/Dockerfile.terminus
+# Terminus 4 drops support for PHP 8.1 which we still need
+ARG TERMINUS_VERSION="3.6.2"
+RUN curl -L --fail -o /usr/local/bin/terminus https://github.com/pantheon-systems/terminus/releases/download/${TERMINUS_VERSION}/terminus.phar && chmod +x /usr/local/bin/terminus
+```
+
+`terminus` is just an example here, it could be any command you wish, 
+[either because you are running an older PHP version](https://github.com/pantheon-systems/terminus/releases/tag/4.0.0)
+or the bundled version [has a bug that ruins things for you](https://github.com/platformsh/cli/discussions/166).
+
+## Installing custom tools
+
+You can obviously use the same trick to install custom tools:
+
+```dockerfile
+# .ddev/web-build/Dockerfile.fzf
+# fooscript relies on fzf
+ARG FZF_VERSION="0.62.0"
+RUN curl -s -L https://github.com/junegunn/fzf/releases/download/v${FZF_VERSION}/fzf-${FZF_VERSION}-linux_amd64.tar.gz | tar xvz -C /usr/local/bin/ && chmod +x /usr/local/bin/fzf
+```
```

Comment:
Add a Resources section pointing to 
* https://ddev.com/blog/customizing-ddev-local-images-with-a-custom-dockerfile/
* https://ddev.com/blog/ddev-docker-image-maintenance/
* Key docs in ddev.readthedocs.io
* Anything else that helped you.

And add a CTA at the end like some of the other blogs telling how to get support and how to contribute. In this case also invite guest blogs.